jax/lax/lax.py

```diff
@@ -4882,6 +4882,10 @@ def _reduce_window_shape_rule(operand, init_value, *, jaxpr, consts,
     msg = ("reduce_window got inconsistent dtypes for operand and init_value: "
            " got operand dtype {} and init_value dtype {}.")
     raise TypeError(msg.format(operand.dtype, init_value.dtype))
+  if init_value.shape != ():
+    msg = ("reduce_window expected init_value to have shape () but init_value "
```

Comment:
nit: maybe say "reduce_window expected init_value to be a scalar"

Reply:
I was a bit hesitant about that, because depending on context we might expect shape `(1,)` to correspond to a scalar (I think?), but in this case, only `()` is acceptable. WDYT?

Reply:
I don't think `(1,)` is ever considered a scalar, but I see how that could be confusing. I'm fine with either then. The reason I suggested saying "a scalar" is because I think it reads more clearly and makes it more clear it _always_ needs to be a scalar and doesn't depend on other circumstances (I feel like `()` is used more when you're printing a shape that may be something else too).

Reply:
After chatting offline, decided to make this change!